openedx/features/qverse_features/registration/models.py

```diff
@@ -101,8 +108,43 @@ def save(self, *args, **kwargs):
 
         This would help us to ensure unique constraint of registration number.
         """
+        # Our prospective user has become regular user
+        # so it's unnecessary to keep its details in two
+        # different tables
+        ProspectiveUser.objects.get(registration_number=self.registration_number).delete()
         self.registration_number = self.registration_number.upper()
-        return super(QVerseUserProfile, self).save(*args, **kwargs)
+        super(QVerseUserProfile, self).save(*args, **kwargs)
 
     def __str__(self):
         return '({}) --- {}'.format(self.registration_number, self.user.profile.name)
+
+
+class ProspectiveUser(models.Model):
+    """
+    Represents qverse prospective user.
+    """
+
+    class Meta(object):
+        app_label = APP_LABEL
+
+    email = models.EmailField(unique=True, validators=[EmailValidator])
+    first_name = models.CharField(max_length=FIRST_NAME_MAX_LENGTH)
+    surname = models.CharField(max_length=SURNAME_MAX_LENGTH)
+    department = models.ForeignKey(Department, db_index=True, null=True, on_delete=models.SET_NULL)
+    registration_number = models.CharField(unique=True, max_length=REGISTRATION_NUMBER_MAX_LENGTH, db_index=True)
+    other_name = models.CharField(blank=True, max_length=OTHER_NAME_MAX_LENGTH)
+    mobile_number = models.CharField(blank=True, max_length=20)
+    current_level = models.IntegerField(choices=LEVEL_CHOICES,
+                                        validators=[validate_current_level, MaxValueValidator(99)])
+    programme = models.IntegerField(db_index=True, choices=PROGRAMME_CHOICES, validators=[validate_programme_choices])
+
+    def __str__(self):
+        return '({}) --- {} {}'.format(self.registration_number, self.first_name, self.surname)
+
+    def save(self, *args, **kwargs):
+        """
+        Make sure that no invalid email is saved in model.
+        """
+        validate_email = EmailValidator()
+        validate_email(self.email)
```

Comment:
Can we use a `clean_email` or `clean` method for the purpose ?

Reply:
It didn't work the way I needed it.